indy_node/server/request_handlers/domain_req_handlers/nym_handler.py

```diff
@@ -83,11 +83,29 @@ def update_state(self, txn, prev_result, request, is_committed=False):
         return existing_data
 
     def _validate_new_nym(self, request, operation):
+        identifier, req_id, _ = get_request_data(request)
         role = operation.get(ROLE)
-        self.write_req_validator.validate(request,
-                                          [AuthActionAdd(txn_type=NYM,
-                                                         field=ROLE,
-                                                         value=role)])
+
+        nym_data = self.database_manager.idr_cache.getNym(request.identifier, isCommitted=False)
+        if not nym_data:
+            # Non-ledger nym case. These two checks duplicated and mainly executed in client_authn,
+            # but it has point to repeat them here, for clear understanding of validation non-ledger request cases.
+            if request.identifier != request.operation[TARGET_NYM]:
```

Comment:
Should we call `request.operation.get(TARGET_NYM)`?

Reply:
ok